plugins/dotplot-view/src/DotplotDisplay/index.ts

```diff
@@ -71,19 +72,25 @@ export function stateModelFactory(configSchema: any) {
 
       return {
         afterAttach() {
-          makeAbortableReaction(
-            self as any,
-            renderBlockData,
-            renderBlockEffect as any,
-            {
-              name: `${self.type} ${self.id} rendering`,
-              delay: 1000,
-              fireImmediately: true,
-            },
-            this.setLoading,
-            this.setRendered,
-            this.setError,
-          )
+          const parent = getContainingView(self) as DotplotViewModel
+          autorun(reaction => {
+            if (parent.initialized) {
+              makeAbortableReaction(
+                self as any,
+                renderBlockData,
+                renderBlockEffect as any,
+                {
+                  name: `${self.type} ${self.id} rendering`,
+                  delay: 1000,
+                  fireImmediately: true,
+                },
+                this.setLoading,
+                this.setRendered,
+                this.setError,
+              )
+              reaction.dispose()
+            }
+          })
```

Comment:
This nested autorun-reaction thing is a little frightening, I'm not sure what it might do in corner cases.

What if you added the `if(parent.initialized)` logic to the renderBlockData and renderBlockEffect functions instead?

Reply:
I'll try that out